update/Pos/Update/Poll/PollState.hs

```diff
@@ -0,0 +1,89 @@
+       , psDelActivePropsIdx
+       , psSlottingData
+       , psFullRichmenData
+       , psIssuersStakes
+
+       -- * Functions
+       , modifyPollState
+       ) where
+
+import           Universum
+
+import           Control.Lens                 (makeLenses)
+import           Data.HashSet                 (HashSet)
+import qualified Data.HashMap.Strict          as HM
+
+import           Pos.Core.Types               (ApplicationName, BlockVersion,
+                                               BlockVersionData, EpochIndex,
+                                               NumSoftwareVersion, SoftwareVersion,
+                                               StakeholderId)
+import           Pos.Lrc.DB.Issuers           (IssuersStakes)
+import           Pos.Lrc.Types                (FullRichmenData)
+import           Pos.Slotting.Types           (SlottingData)
+import           Pos.Update.Core              (UpId)
+import           Pos.Update.Poll.Types        (BlockVersionState, ConfirmedProposalState,
+                                               PollModifier (..), ProposalState)
+import           Pos.Util.Modifier            (deletions, insertions)
+
+data PollState = PollState
+    { -- | All competing block versions with their states
+      _psBlockVersions      :: !(HashMap BlockVersion BlockVersionState)
+      -- | Presently adopted block version with its data
+    , _psAdoptedBV          :: !(BlockVersion, BlockVersionData)
+      -- | Software version of this node
+    , _psSoftwareVersion    :: !SoftwareVersion
+      -- | All stakeholders who made proposals in the current epoch
+    , _psEpochProposers     :: !(HashSet StakeholderId)
+      -- | All applications in use and their latest (confirmed) versions
+    , _psConfirmedBVs       :: !(HM.HashMap ApplicationName NumSoftwareVersion)
+      -- | All proposed software versions and their state
+    , _psConfirmedProposals :: !(HM.HashMap SoftwareVersion ConfirmedProposalState)
+      -- | All update proposals and their states
+    , _psActiveProposals    :: !(HM.HashMap UpId ProposalState)
+      -- | Update proposals for each application
+    , _psDelActivePropsIdx  :: !(HM.HashMap ApplicationName (HashSet UpId))
+      -- | Slotting data for this node
+    , _psSlottingData       :: !SlottingData
+      -- | Mapping between epochs and their richmen stake distribution
+    , _psFullRichmenData    :: !(HM.HashMap EpochIndex FullRichmenData)
+      -- | Mapping between epochs and stake of each of the epoch's slot's block issuer
+    , _psIssuersStakes      :: !(HM.HashMap EpochIndex IssuersStakes)
+    } deriving (Show, Eq)
+
+makeLenses ''PollState
+
+modifyPollState :: PollModifier -> PollState -> PollState
+modifyPollState PollModifier {..} PollState {..} =
+    let insAndDel pm hm =
+            foldl' (flip (uncurry HM.insert)) (foldl' (flip HM.delete) hm deletes) inserts
+          where inserts = insertions pm
+                deletes = deletions pm
```

Comment:
I think this function (`insAndDel :: MapModifier k v -> HashMap k v -> HashMap k v`) should be in `Pos.Util.Modifier`. I would call it `applyModifierToHashMap` or `modifyHashMap` or `applyToHashMap`.